### PR TITLE
[8.0][OAuth 2] dirac-login try to connect to DIRAC AS if no local certificate

### PR DESCRIPTION
**Pull request by TaykYoku**
1.

Show infomation message:
```
You are currently using proxy/token to access new HTTP DIRAC services....
```
only if the current state of the user environment does not match the authorization result

2.

### Old logic:

No local certificate:
```bash
(dirac-development) Armstrongs-MacBook-Pro:DIRAC TaykYoku$ dirac-login
Can't find user certificate and key
```

### With this change:

No local certificate:
```bash
(dirac-development) Armstrongs-MacBook-Pro:DIRAC TaykYoku$ dirac-login
Can't find user certificate and key, trying to connact to DIRAC AS..
Use next link to continue
https://dirac.egi.eu/auth/device?user_code=RWKG-MBCK
Authorization pending.. (use CNTL + C to stop)
^CUser canceled the operation..
```

With local certificate:
```bash
(dirac-development) Armstrongs-MacBook-Pro:DIRAC TaykYoku$ dirac-login
Enter Certificate password:
Upload proxy to server.
```

BEGINRELEASENOTES

*Framework
NEW: dirac-login try to connect to DIRAC AS if no local certificate

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/FrameworkSystem/scripts/dirac_login.py [hunk 1]
```diff
@@ -261,9 +264,12 @@ def loginWithCertificate(self):
         """Login with certificate"""
         # Search certificate and key
         if not self.certLoc or not self.keyLoc:
-            cakLoc = Locations.getCertificateAndKeyLocation()
-            if not cakLoc:
-                return S_ERROR("Can't find user certificate and key")
+            if not (cakLoc := getCertificateAndKeyLocation()):
+                problem = "Can't find user certificate and key"
+                if not self.authWith:  # if user do not choose this way
+                    print(HTML(f"<yellow>{problem}</yellow>" + ", trying to connact to DIRAC AS.."))
```
Comment (fstagni) [inline src/DIRAC/FrameworkSystem/scripts/dirac_login.py:270]:
```suggestion
                    print(HTML(f"<yellow>{problem}</yellow>" + ", trying to connect to DIRAC AS.."))
```
and do you really need to define a `problem` "variable"?

File: src/DIRAC/FrameworkSystem/scripts/dirac_login.py [hunk 2]
```diff
@@ -37,6 +37,9 @@
     getTokenFileLocation,
 )
 
+DEFAULT_AUTH_WAY = "certificate"
+DEFAULT_RESPONSE = "proxy"
```
Comment (fstagni) [inline src/DIRAC/FrameworkSystem/scripts/dirac_login.py:41]: What's possible (the strings "certificate", "proxy", "token" etc) should be defined in a separate module IMHO, as these strings IIUC will certainly be used elsewhere too.

File: src/DIRAC/FrameworkSystem/scripts/dirac_login.py (hunk 1 above)
Comment (TaykYoku) [reply src/DIRAC/FrameworkSystem/scripts/dirac_login.py]: problem "variable" removed

File: src/DIRAC/FrameworkSystem/scripts/dirac_login.py (hunk 2 above)
Comment (TaykYoku) [reply src/DIRAC/FrameworkSystem/scripts/dirac_login.py]: I added comments about possible variables

These values are tied to the keys used by the user, so I defined the default values here and this way. I think that if there will be a precedent for the wider use of these values, then they can be defined in another, more appropriate place

Comment (fstagni) [reply src/DIRAC/FrameworkSystem/scripts/dirac_login.py]: I was more thinking of adding a file like https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/Core/Security/Properties.py

Now I merge it but please add it to another PR.

**Pull request merged**